### PR TITLE
Include instance realm and version in various events/posthog.people.set calls

### PR DESCRIPTION
**Pull request by macobo**
Related issue: https://github.com/PostHog/company-internal/issues/313

Aside: It's kind of wierd how few person properties we're setting. Is
that intentional?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [ ] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

**Comment by kpthatsme** [comment]
Is `hosted-clickhouse` === scale? Or does it represent both scale and scale-free?

I see this check– not sure if scale free will have this HELM info set or not:

```    
def get_instance_realm() -> str:
    """
    Returns the realm for the current instance. `cloud` or `hosted` or `hosted-clickhouse`.
    """
    if settings.MULTI_TENANCY:
        return "cloud"
    elif os.getenv("HELM_INSTALL_INFO", False):
        return "hosted-clickhouse"
    else:
        return "hosted"

**Comment by macobo** [comment]
It represents hosted clickhouse - that's both scale and scale-free since the two are not differentiated at this time anywhere in the codebase.

**Comment by kpthatsme** [comment]
> It represents hosted clickhouse - that's both scale and scale-free since the two are not differentiated at this time anywhere in the codebase.

Okay thanks good to know.

**Review by kpthatsme — APPROVED** [review]

File: posthog/api/user.py [hunk 1]
```diff
@@ -289,6 +290,7 @@ def user(request):
             "is_staff": user.is_staff,
             "is_impersonated": is_impersonated_session(request),
             "is_event_property_usage_enabled": getattr(settings, "ASYNC_EVENT_PROPERTY_USAGE", False),
+            "realm": get_instance_realm(),
```
Comment (paolodamico) [inline posthog/api/user.py:293]: This is only updating a deprecated endpoint. We shouldn't be sending this here.

File: posthog/api/test/test_preflight.py [hunk 2]
```diff
@@ -53,6 +54,7 @@ def test_preflight_request(self):
                     "db": True,
                     "initiated": True,
                     "cloud": False,
+                    "realm": "hosted",
```
Comment (paolodamico) [inline posthog/api/test/test_preflight.py:57]: I think it's worth adding is a test case for `hosted-clickhouse`, when do we send this attribute?

File: frontend/src/types.ts [hunk 3]
```diff
@@ -41,6 +41,8 @@ export interface UserType {
     organization: OrganizationType | null
     team: TeamBasicType | null
     organizations: OrganizationBasicType[]
+    realm: 'cloud' | 'hosted' | 'hosted-clickhouse'
```
Comment (paolodamico) [inline frontend/src/types.ts:44]: This is an instance-based property, not a user-based property, I don't think we should send this in the user payload at all. See also comment about sending this in the deprecated endpoint.

File: posthog/views.py [hunk 4]
```diff
@@ -87,6 +88,7 @@ def preflight_check(request: HttpRequest) -> JsonResponse:
         "db": is_postgres_alive(),
         "initiated": User.objects.exists() if not settings.E2E_TESTING else False,  # Enables E2E testing of signup flow
         "cloud": settings.MULTI_TENANCY,
+        "realm": get_instance_realm(),
```
Comment (paolodamico) [inline posthog/views.py:91]: Should we get rid of the `cloud` property? I'm worried about having more than one source of truth for something. Will cause headaches down the line.

File: frontend/src/scenes/userLogic.tsx [hunk 5]
```diff
@@ -78,7 +78,11 @@ export const userLogic = kea<userLogicType<UserType, UpdateUserPayload>>({
                     }
 
                     posthog.identify(user.distinct_id)
-                    posthog.people.set({ email: user.anonymize_data ? null : user.email })
+                    posthog.people.set({
+                        email: user.anonymize_data ? null : user.email,
+                        realm: user.realm,
```
Comment (paolodamico) [inline frontend/src/scenes/userLogic.tsx:83]: We already set this information in the `preflightLogic` (see https://github.com/PostHog/posthog/blob/27144478f534290ee01b69d6ac5c6a53b8e05ec4/frontend/src/scenes/PreflightCheck/logic.ts#L80).

File: frontend/src/scenes/userLogic.tsx [hunk 6]
```diff
@@ -78,7 +78,11 @@ export const userLogic = kea<userLogicType<UserType, UpdateUserPayload>>({
...
+                        posthog_version: user.posthog_version,
```
Comment (paolodamico) [inline frontend/src/scenes/userLogic.tsx:84]: Same as above

File: frontend/src/scenes/PreflightCheck/logic.ts [hunk 7]
```diff
@@ -36,11 +36,11 @@ export const preflightLogic = kea<preflightLogicType<PreflightStatus, PreflightM
         ],
         realm: [
             (s) => [s.preflight],
-            (preflight): 'cloud' | 'hosted' | null => {
+            (preflight): 'cloud' | 'hosted' | 'hosted-clickhouse' | null => {
```
Comment (paolodamico) [inline frontend/src/scenes/PreflightCheck/logic.ts:39]: Maybe it's better to get rid of the selector altogether and use the value inside the `preflight` object instead, avoid duplicate information.

**Review by paolodamico — COMMENTED** [review]
Sorry I'm just getting to a review here, but I do think there are a few things worth addressing.

Also, the reason you're seeing so little super properties being set is that all instance-based properties are set in the preflight logic file.

**Pull request merged**